File: C++/First Steps in Coding - Basics of C++/While Loops/Exercise/Walking.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
	int stepsGoal = 10000, totalSteps=0;
	bool goalReached = false;
	while (true) {
		string steps;
		getline(cin, steps);
		if (steps != "Going home") {
			int stepsDone = stoi(steps);
			totalSteps += stepsDone;
			if (totalSteps >= stepsGoal) {
				goalReached = true;
				break;
			}
		}
		else {
			int stepsOnWayHome;
			cin >> stepsOnWayHome;
			totalSteps += stepsOnWayHome;
			if (totalSteps >= stepsGoal) {
				goalReached = true;
				break;
			}
			else {
				break;
			}
		}
	}

	if (goalReached == false) {
		cout << stepsGoal - totalSteps << " more steps to reach goal.";
	}
	else {
		cout << "Goal reached! Good job!";
	}
}

File: Python/Basics/While Loop/Exercise/Walking.py
totalSteps = 0
goalSteps = 10000
going = True

while True:
    dailySteps = input()

    if dailySteps != "Going home":
        dailySteps = int(dailySteps)
        totalSteps += dailySteps

    else:
        stepsToHome = int(input())
        totalSteps += stepsToHome
        going = False

    if totalSteps >= goalSteps:
        print(f"Goal reached! Good job!\n{totalSteps - goalSteps} steps over the goal!")
        break
    else:
        if not going:
            print(f"{goalSteps - totalSteps} more steps to reach goal.")
            break
